### PR TITLE
dlab-11 add comment about credentials

### PR DESCRIPTION
**Pull request by hemeshpatel-moj**

### CODE REVIEW

**Review by PriyaBasker23 — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  preview:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.10.10]
+    steps:
+      - uses: actions/checkout@v2
```
Comment (jacobwoffenden) [inline .github/workflows/test.yaml:14]:
```suggestion
      - uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          fetch-depth: 1
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
```
Comment (jacobwoffenden) [inline .github/workflows/test.yaml:19]:
```suggestion
        uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435 # v4.5.0
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python dependencies
+        run: |
+          
+          pip install -r data-products/_example_2/requirements.txt
```
Comment (jacobwoffenden) [inline .github/workflows/test.yaml:26]:
```suggestion
        run: |
          pip install -r data-products/_example_2/requirements.txt
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: [3.10.10]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
```
Comment (jacobwoffenden) [inline .github/workflows/test.yaml:16]: Is this required?

File: .github/workflows/test.yaml
Comment (jacobwoffenden) [inline .github/workflows/test.yaml:1]: Can this be renamed to something more descriptive?

`data-product-example-2-test.yml`

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: .github/workflows/data-product-example-2-test.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+name: Data Product - example_2 - Test
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - .github/workflows/<this workflow file>
```
Comment (jacobwoffenden) [inline .github/workflows/data-product-example-2-test.yaml:9]:
```suggestion
      - .github/workflows/data-product-example-2-test.yaml
```

File: .github/workflows/data-product-example-2-test.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/setup-python@d27e3f3d7c64b4bbf8e4abfb9b63b83e846e0435 # v4.5.0
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Python dependencies
+        run: |
+          pip install --requirement requirements.txt
+      - name: Run tests with pytest
+        run: |
+          pytest tests/ -vv;
```
Comment (jacobwoffenden) [inline .github/workflows/data-product-example-2-test.yaml:36]:
```suggestion
      - name: Install Python dependencies
        run: |
          pip install --requirement requirements.txt

      - name: Run tests with pytest
        run: |
          pytest tests/ -vv;
```

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**